GCRCatalogs/register.py

```diff
@@ -126,12 +126,12 @@ def _get_site_info(self):
         Return a string which, when executing at a recognized site with
         well-known name, will include the name for that site
         """
-        # First look for custom env variable
-        v = os.getenv(self._DESC_SITE_ENV)
-        if v:
-            warnings.warn("Site determined from env variable {}".format(self._DESC_SITE_ENV))
-            return v
-        return socket.getfqdn()
+        site_from_env = os.getenv(self._DESC_SITE_ENV)
+        site_from_socket = socket.getfqdn()
+        if site_from_env and site_from_socket and site_from_env not in site_from_socket:
+            warnings.warn("Site determined from env variable {} = {}".format(self._DESC_SITE_ENV, site_from_env))
+            return site_from_env
+        return site_from_socket
```

Comment:
I agree there should be no warning in case `site_from_env` is not empty and is in `site_from_socket` but, to make this more foolproof I prefer
* if `site_from_env` has a value but `site_from_socket` doesn't, return `site_from_env` (just in case it's possible for socket.getfqdn() to return None or empty string)
* they both have a value and `site_from_env  in site_from_socket`, return `site_from_env` (without any warning) rather than `site_from_socket`.  We know the return value from getfqdn( ) includes extraneous stuff, conceivably it can include more than one legitimate site value.  It's not likely, but I think it's even less likely that someone would provide a value in the environment value with this problem.

Reply:
Thanks. That sounds reasonable. I'll update the logic here.